nomos-da/kzgrs-backend/src/encoder.rs

```diff
@@ -55,6 +56,65 @@ pub struct EncodedData {
     pub column_commitments: Vec<Commitment>,
     pub aggregated_column_commitment: Commitment,
     pub aggregated_column_proofs: Vec<Proof>,
+
+    /// Iterator State
+    current_column_idx: Option<usize>,
```

Comment:
This is not the way to go. Usually what you want to do is to implement `IntoIterator` then return a new struct that handles this state and have a reference to the original Encoded data and that struct implements `Iterator`.

Reply:
Oh, right! This actually makes waaaay more sense. I didn't have a great feeling about it, but as everyone was suggesting doing it this way I assumed it was the standard.

Thanks for pointing this out! Updating